src/state/tls.py

```diff
@@ -22,29 +30,36 @@ class TLSInformation:
 
     Attrs:
         secret_resource_name_prefix: Prefix of the secret resource name.
-        tls_requirer_integration: The integration instance with a TLS provider.
         tls_certs: A dict of hostname: certificate obtained from the relation.
         tls_keys: A dict of hostname: private_key stored in juju secrets.
     """
 
     secret_resource_name_prefix: str
-    tls_requirer_integration: Relation
     tls_certs: dict[str, str]
     tls_keys: dict[str, dict[str, str]]
 
     @classmethod
-    def from_charm(cls, charm: ops.CharmBase) -> "TLSInformation":
+    def from_charm(
+        cls, charm: ops.CharmBase, certificates: TLSCertificatesRequiresV3
+    ) -> "TLSInformation":
         """Get TLS information from a charm instance.
 
         Args:
             charm: The gateway-api-integrator charm.
+            certificates: TLS certificates requirer library.
 
         Raises:
             TlsIntegrationMissingError: When integration is not ready.
+            SecretNotSupportedError: When running with incompatible juju version.
 
         Returns:
             TLSInformation: Information about configured TLS certs.
         """
+        if not JujuVersion.from_environ().has_secrets:
+            raise SecretNotSupportedError(
+                "The charm requires the 'secrets' feature to be supported (juju version >= 3.0.3)."
+            )
```

Comment:
I don't think this is necessary since we set:
https://github.com/canonical/gateway-api-integrator-operator/blob/52b18b5c0a5f1c5002d9977854b815f6aa491ae8/charmcraft.yaml#L36

Meaning that Juju won't let the charm being deployed if the controller is not >3.3